helix-term/src/commands.rs

```diff
@@ -1941,6 +1955,155 @@ mod cmd {
         Ok(())
     }
 
+    fn debug_eval(
+        cx: &mut compositor::Context,
+        args: &[&str],
+        _event: PromptEvent,
+    ) -> anyhow::Result<()> {
+        use helix_lsp::block_on;
+        if let Some(debugger) = cx.editor.debugger.as_mut() {
+            let (frame, thread_id) = match (debugger.active_frame, debugger.thread_id) {
+                (Some(frame), Some(thread_id)) => (frame, thread_id),
+                _ => {
+                    bail!("Cannot find current stack frame to access variables")
+                }
+            };
+
+            // TODO: support no frame_id
+
+            let frame_id = debugger.stack_frames[&thread_id][frame].id;
+            let response = block_on(debugger.eval(args.join(" "), Some(frame_id)))?;
+            cx.editor.set_status(response.result);
+        }
+        Ok(())
+    }
+
+    fn edit_breakpoint_impl(
+        cx: &mut compositor::Context,
+        condition: Option<String>,
+        log_message: Option<String>,
+    ) -> anyhow::Result<()> {
+        use helix_lsp::block_on;
+
+        let (view, doc) = current!(cx.editor);
+        let text = doc.text().slice(..);
+        let pos = doc.selection(view.id).primary().cursor(text);
+        let breakpoint = helix_dap::SourceBreakpoint {
+            line: text.char_to_line(pos) + 1, // convert from 0-indexing to 1-indexing (TODO: could set debugger to 0-indexing on init)
+            condition,
+            log_message,
+            ..Default::default()
+        };
+        let path = match doc.path() {
+            Some(path) => path.to_path_buf(),
+            None => {
+                bail!("Can't edit breakpoint: document has no path")
+            }
+        };
+
+        let breakpoints = cx.editor.breakpoints.entry(path.clone()).or_default();
+        if let Some(pos) = breakpoints.iter().position(|b| b.line == breakpoint.line) {
+            breakpoints.remove(pos);
+            breakpoints.push(breakpoint);
+
+            let breakpoints = breakpoints.clone();
+
+            if let Some(debugger) = &mut cx.editor.debugger {
+                // TODO: handle capabilities correctly again, by filterin breakpoints when emitting
```

Comment:
I'll handle this when dealing with breakpoint feedback from debugger (unverified etc)

Reply:
We will split breakpoints into "wanted" and "real". User edits "wanted" ones, but sees whether current debugger can make them "real" (or set their properties).